src/swift/Sources/Models/TokenFile.swift

```diff
@@ -168,59 +183,49 @@ class TokenFile: Codable {
         text(packageName)
         whitespace()
         punctuation("{")
+        newLine()
         indent {
             declarations.forEach { topLevelDecl in
                 process(topLevelDecl)
             }
         }
-        if needsNewLine {
-            newLine()
-        }
         punctuation("}")
         newLine()
 
-        navigation(from: declarations)
+        navigationTokens(from: declarations)
         // ensure items appear in sorted order
-        navigation.sort(by: {$0.text < $1.text })
-        for item in navigation {
+        navigation.sort(by: { $0.text < $1.text })
+        navigation.forEach { item in
             item.childItems.sort(by: { $0.text < $1.text })
         }
     }
 
-    private func process(_ decl: TopLevelDeclaration) {
-        if needsNewLine {
-            newLine()
-        }
-        for statement in decl.statements {
+    private func process(_ decl: TopLevelDeclaration, overridingAccess: AccessLevelModifier? = nil) {
```

Comment:
What's the purpose of `overridingAccess`?

Reply:
For the most part, if something doesn't have an access modifier, it is considered `internal`. However, there are very specific cases where this is not true. The members of a `public` protocol, for example, will always be public. Thus, a declaration may not have an access modifier but may still need to be displayed depending on its parent's access.

https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html

Reply:
Ok thank you that makes a lot of more sense